### PR TITLE
34 Allow specification of entrypoints in workflow dispatch

### LINKED ISSUE #34: Allow specification of entrypoints in workflow dispatch
When running with workflow dispatch, we probably want to have different entry points that might be triggered for the nextflow workflow (test, simulate, etc). We will add these as inputs to the workflow dispatch trigger and then use them as needed in the action.

### PR DESCRIPTION
**Pull request by jashapiro**
closes #34

Here I am adding a dropdown to the workflow menu to allow running the workflow with various entrypoints. 

These are:
-`test`: runs the minimal test workflow which only tests infrastructure, though we could potentially modify this to run other tests in the future...
- `simulated`: runs the simulation workflow first, then runs the main workflow on the simulated data to generate simulated results
- `full` (don't love this name... maybe "release"?): Runs the full workflow with real data.

Some thoughts on reflection, for which I would like input: 
- Maybe I should split off the actual simulation and only run that on request?
- Alternatively, should the "release" or "full" version run twice: once with the simulated data and once with the real? Part of me things that is the right move, as we would almost always want to keep those in sync.

Note that I will include docs for these options in #32

I will not merge this until I have had a chance to test #39, which is currently held up by GHA being "degraded"

### CODE REVIEW

File: scripts/run_nextflow.sh [hunk 1]
```diff
@@ -3,24 +3,61 @@
...
 cd /opt/nextflow
 nextflow pull AlexsLemonade/OpenScPCA-nf -revision $GITHUB_TAG
 
+# test mode runs the test workflow only, then exits
+if [ "$RUN_MODE" == "test" ]; then
+  nextflow run AlexsLemonade/OpenScPCA-nf \
+    -revision $GITHUB_TAG \
+    -entry test \
+    -profile $profile \
+    -entry test \
```
Comment (jaclyn-taroni) [inline scripts/run_nextflow.sh:21]: I assume?
```suggestion
```

File: scripts/run_nextflow.sh [hunk 2]
```diff
@@ -3,24 +3,61 @@
...
+  # Copy logs to S3 and delete if successful
+  aws s3 cp . s3://openscpca-nf-data/logs/${RUN_MODE}/${date} \
+    --recursive \
+    --exclude "*" \
+    --include "${datetime}_*" \
+    && rm ${datetime}_*
+  exit 0
+fi
+
+# for simulated mode, run the data simulation pipeline first
```
Comment (jaclyn-taroni) [inline scripts/run_nextflow.sh:36]: So if you select simulated, the full will also run? I'm not sure I would expect that. (I understand docs are coming!)

Comment (jaclyn-taroni) [reply scripts/run_nextflow.sh:36]: I went back and re-read your comments. I'm leaving this here so you have my genuine reaction. 😄 

I agree that we probably want to keep these in sync; the use of `simulated` would make me think it's just simulated.

I wonder if it should be something like: `simulated` (simulated only), `release` (ScPCA data – I also thought `scpca` might work), and `full` (simulated and ScPCA data, where you'd set this as the default.)

**Review by jaclyn-taroni — COMMENTED** [review]
Returning my input! And one bug, I think.

File: scripts/run_nextflow.sh (hunk 2 above)
Comment (jashapiro) [reply scripts/run_nextflow.sh:35]: Okay, I think I will go with `simulated`, `scpca`, and `full`, but in a change from the current setup, the _simulation_ step will only run with `full`, which is what will be triggered with a workflow release. 

Which is to say: 
-`simulated`: run the main workflow with simulated data
-`scpca`: run the main workflow with real data
-`full`: simulate data based on real data, then run the main workflow once with the simulated data and once with the real data
-`test`: same as before

Should I also have a `simulate-only`? Or am I wrong and the `simulated` workflow should also run the simulation component?

Comment (jaclyn-taroni) [reply scripts/run_nextflow.sh:35]: > -`simulated`: run the main workflow with simulated data
> -`scpca`: run the main workflow with real data
> -`full`: simulate data based on real data, then run the main workflow once with the simulated data and once with the real data
> -`test`: same as before

I think this is right

**Comment by jashapiro** [comment]
Now that I got things working again, I have updated with new options as we discussed in the comments. Should be ready for review!

**Review by jaclyn-taroni — APPROVED** [review]
Returning a bug fix. LGTM otherwise 👍🏻

**Pull request merged**